src/api/java/com/minecolonies/api/research/AbstractResearchProvider.java

```diff
@@ -360,7 +383,54 @@ public Research addBuildingRequirement(final String buildingName, final int leve
                 reqArray = new JsonArray();
             }
             JsonObject req = new JsonObject();
-            req.addProperty("building", buildingName);
+            // Temporary workaround for the discrepancy between schematic ID and modBuildings ID.
```

Comment:
Consider extracting this into a helper function rather than repeating it in multiple places.